src/model.rs

```diff
@@ -894,50 +900,66 @@ impl Collections {
                 stop_areas.sort_by_key(|stop_area| &stop_area.name);
                 stop_areas.get(0).cloned()
             }
-            fn find_stop_area_for<'a, F>(
+            fn find_best_stop_area_for<'a, F>(
                 collections: &'a Collections,
-                route_id: &'a str,
+                vehicle_journey_idxs: &IdxSet<VehicleJourney>,
                 select_stop_point_in_vj: F,
             ) -> Option<&'a StopArea>
             where
                 F: Fn(&VehicleJourney) -> Idx<StopPoint>,
             {
                 let stop_areas: Vec<Idx<StopArea>> =
-                    select_stop_areas(collections, route_id, select_stop_point_in_vj);
+                    select_stop_areas(collections, vehicle_journey_idxs, select_stop_point_in_vj);
                 let by_frequency: HashMap<Idx<StopArea>, usize> = group_by_frequencies(stop_areas);
```

Comment:
minor (feel free to disregard and resolve, especially as it's not the point of your PR and you've done enough!):
creating a `fn cmp()` on stop_areas here would seem more idiomatic.

Maybe even impl `std::cmp::Ord` for stop_areas, but I'm not sure it makes sense for all situations where stop_areas would be compared.

Reply:
I thought about that but reach the same conclusion, I'm not sure it makes any sense to compare `StopArea`, what does it even mean? And because I was not able to answer that question in every situation, I decided that `std::cmp::Ord` was not the way to go.